content/blog/backstage-fargate-up-and-running/index.md

```diff
@@ -0,0 +1,346 @@
+
+
+
+## Step 3 - Defining our environment in System Manager Parameter Store
+
+We know what the connection string is to connect to our database so next we will move on to set up those. We'll also set up our  Github token in the same way as an environment variable (make sure you have created a Github token explained [here](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token), if you want to use Github with your Backstage). There are few different ways to pass in environment variables to running containers in AWS ECS. We will be using AWS System Manager Parameter Store to save those in a safe place where they can be then loaded to the running container. AWS ECS also provides the possibility to load environment variables from a flat file stored in S3 or pass them in directly (unsecured) to the task definition. 
+
+Let's navigate to the [Parameter Store](./https://eu-west-1.console.aws.amazon.com/systems-manager/parameters/?region=eu-west-1&tab=Table) and populate the needed values in there.
+By clicking Create Parameter we can create values for our database credentials for the RDS instance we created previously as well as the Github token we have created.
+
+![param_store_create.png](./param_store_create.png)
+
+At least the `DB_PASSWORD` and `GITHUB_TOKEN` should be of type SecureString, so they are encrypted. We'll be using just the default KMS Key in this case to encrypt the values, but it might be worthwhile to generate a specific key for these parameters. 
+
+
+In the end we should be ending up with a list of few parameters that we can use later. 
+
+![param_store_list.png](./param_store_list.png)
+
+Note that we have not defined the database port to be retrieved from parameter store here. That might be something you want to do if the ports change regularly or are non-standard, but is not really necessary. 
+
+
+## Step 4 - Creating a Load Balancer for our Backstage service
+
+The last scaffolding bit we want to do to support our Fargate Backstage is to set up a load balancer in standby to wait for our Fargate service to attach itself to it. We do this step a bit prematurely just to have a good static URL available to point to when we eventually start building the actual Backstage application. 
+
+Let's navigate to [AWS Load Balancer section](https://eu-west-1.console.aws.amazon.com/ec2/v2/home?region=eu-west-1#LoadBalancers) in the console and spin one up. 
+
+We want to create an Application Load balancer that is internet facing. It is a good idea to select all subnets, so the load balancer is able to target our containers even if they are spread out across different AWS Availability Zones. For now the only listener we attach to the load balancer is listening to HTTP traffic through port 80, but if you have a domain that you control and can create certificates for, you should be using HTTPS and port 443.  
+![lb_basic_setup.png](./lb_basic_setup.png)
```

Comment:
This needs a newline just above. Without it the image is squashed up against the paragraph above.